authentication/authentication_option.go

```diff
@@ -74,3 +74,20 @@ func WithAuth0ClientEnvEntry(key string, value string) Option {
 		}
 	}
 }
+
+// WithRetries configures the management client to only retry under the conditions provided.
+func WithRetries(maxRetries int, statuses []int) Option {
+	return func(a *Authentication) {
+		a.retryStrategy = client.RetryOptions{
+			MaxRetries: maxRetries,
+			Statuses:   statuses,
+		}
+	}
+}
+
+// WithNoRetries configures the management client to only retry under the conditions provided.
+func WithNoRetries() Option {
```

Comment:
How are we solving this in other parts of the public API? Is it always consistent?

Reply:
I'd leant towards `WithNoRetries` mostly because it felt a bit more intuitive/readable to reach for a method to disable it explicitly. Being able to stack them does make it seem a bit more awkward though, but the options passed to `New` should.

It mirrors the `WithNoAuth0ClientInfo` that is used to disable the Auth0-Client header from being sent